src/data/setoid/basic.lean

```diff
@@ -51,6 +51,12 @@ namespace setoid
 @[ext] lemma ext' {r s : setoid α} (H : ∀ a b, r.rel a b ↔ s.rel a b) :
   r = s := ext H
 
+instance fun_like : fun_like (setoid α) α (λ _, α → Prop) :=
```

Comment:
Sorry for the back and forth, but I don't think this is a good idea. We should have either a fun_like instance or `rel`, but not both. I don't think a refactor between the two belongs in this PR.

Reply:
(I think I'm in favor of such a refactor since we have a coe_fn instance for `con`, but it might not be worth it pre-port)

Reply:
reverted. I wonder if CI finds the correct olean with me using `git revert` or we're going to have to rebuild